flintrock/core.py

```diff
@@ -482,6 +483,58 @@ def _run_asynchronously(*, partial_func: functools.partial, hosts: list):
         loop.close()
 
 
+def get_java_major_version(client: paramiko.client.SSHClient):
+    possible_cmds = [
+        "$JAVA_HOME/bin/java -version",
+        "java -version"
+    ]
+
+    for command in possible_cmds:
+        try:
+            output = ssh_check_output(
```

Comment:
It will still work for the current default AMIs because no output means Java 8 gets installed, but it will do unnecessary work if Java 8 is already installed.